### PR TITLE
E_CLASSROOM-337 [FE/BE] Fix change category in add/edit question

### PR DESCRIPTION
**Pull request by CaneteJoash**
### Issue Link
https://framgiaph.backlog.com/view/E_CLASSROOM-337

### Definition of Done
- [x] User can change category of the quiz
- [x] Add/edit question functionality is not broken

### Commands to Run
N/A

### Related PR
BE: https://github.com/framgia/sph-classroom-els-be/pull/79
#### Main note
N/A
#### Pages Affected
- http://localhost:3003/admin/quizzes/1
- http://localhost:3003/admin/quizzes/1/edit

### Scenarios/ Test Cases
- [x] User can change category of the quiz
- [x] Add/edit question functionality is not broken
#### User Interface
- [x] when click the save button the location in quizDetail should update
#### User Experience 
N/A
#### Functionality
- [x] it should return to Quiz Detail and update the location.
### Screenshots
image
image
image
image
image

### CODE REVIEW

File: src/pages/admin/quizzes/QuizDetail/index.js [hunk 1]
```diff
@@ -26,13 +30,29 @@ const QuizDetail = () => {
...
       setQuestions(data.data);
     });
+    
   }, []);
+  
+  useEffect(() => {
+    CategoryApi.getParentCategories(quizInfo?.category_id)
+      .then(({ data }) => {
+        setParentCategories(data);
+      });
```
Comment (karlolee50) [inline src/pages/admin/quizzes/QuizDetail/index.js]: include a .catch statement

File: src/pages/admin/quizzes/QuizDetail/index.js [hunk 2]
```diff
@@ -26,13 +30,29 @@ const QuizDetail = () => {
...
+  }, [quizInfo]);
+  
+  useEffect(() => {
+    parentCategories?.forEach((p, idx) =>
+      idx === 0
```
Comment (karlolee50) [inline src/pages/admin/quizzes/QuizDetail/index.js]:
```suggestion
      !idx
```

File: src/pages/admin/quizzes/QuizDetail/index.module.scss [hunk 3]
```diff
@@ -52,26 +52,17 @@
...
   font-size: $font-size-lg;
   font-weight: $font-weight-regular;
   margin-bottom: 20px;
 }
+
+.quizCategory {
+  color: rgb(129, 139, 144);
+  font-size: $font-size-md;
+  padding-left: 50px;
+}
```
Comment (karlolee50) [inline src/pages/admin/quizzes/QuizDetail/index.module.scss]: add extra line

File: src/pages/admin/quizzes/QuizEdit/index.js [hunk 4]
```diff
@@ -1,30 +1,34 @@
...
+  const [location, setLocation] = useState(null);
+  const [locationPathDisplay, setLocationPathDisplay] = useState('');
   const [quizInfo, setQuizInfo] = useState(null);
   const { categoryId, quizId } = useParams();
   const [questions, setQuestions] = useState(null);
   const [selectedQuestion, setSelectedQuestion] = useState(null);
   const toast = useToast();
+  const [changeCategoryId, setChangeCategoryId] = useState(null);
+  const [parentCategories, setParentCategories] = useState(null);
+  const [saveLocation, setSaveLocation] = useState(false);
```
Comment (karlolee50) [inline src/pages/admin/quizzes/QuizEdit/index.js]: please group together the variables. for example group together the ones using useState, not using useState, etc..

File: src/pages/admin/quizzes/QuizEdit/index.js [hunk 5]
```diff
@@ -37,6 +41,31 @@ const QuizEdit = () => {
     });
   }, []);
 
+  useEffect(() => {
+    if(quizInfo){
+      setChangeCategoryId(quizInfo.category_id);
+      CategoryApi.getParentCategories(quizInfo.category_id)
+        .then(({ data }) => {
+          setParentCategories(data);
+        });
```
Comment (karlolee50) [inline src/pages/admin/quizzes/QuizEdit/index.js]: add a .catch

File: src/pages/admin/quizzes/QuizEdit/index.js [hunk 6]
```diff
@@ -37,6 +41,31 @@ const QuizEdit = () => {
...
+  useEffect(() => {
+    if (saveLocation) {
+      setChangeCategoryId(location?.id);
+      setLocation(null);
+    }
+  }, [location, saveLocation]);
+
+  useEffect(() => {
+    parentCategories?.forEach((p, idx) =>
+      idx === 0
```
Comment (karlolee50) [inline src/pages/admin/quizzes/QuizEdit/index.js]:
```suggestion
      !idx
```

**Review by karlolee50 — CHANGES_REQUESTED** [review]
Review time: 10 minutes
Comment/s: 6

**Comment by catampongan** [comment]
## Working
- [x] Able to change the category of the quiz
- [x] Add/edit button is functional
- [x] Redirected to quizzes after saving the changes

## Comment
- [ ] Root location is not working, please remove root option
------------------------------
task-337
task-337
Task-337-1

--------------------------------
- Total Test Time: 0.5 hr

**Comment by catampongan** [comment]
## Approved PR #152 
- [x] Able to change the category of the quiz
- [x] Add/edit button is functional
- [x] Redirected to quizzes after saving the changes
---------------------------------
task-337
Task-337-1
task-337

-----------------------
- Total Test Time: 0.75 hr
- Total Iterations: 1
- Total Comments: 1

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**